lambdas/aip.py

```diff
@@ -248,9 +248,16 @@ def process_file_worker(row: pd.Series) -> None:
                 f"{self.s3_key}/{filepath}"
             ]
 
-            # retrieve or generate SHA256
-            if inventory_row["checksum_algorithm"] != "SHA256":
+            # derive a SHA256 checksum for file
+            ## case: S3 objects >= 5gb
+            if inventory_row["size"] > 5 * 1024 * 1024 * 1024:  # 5 GB
+                base64_checksum = self.s3_client.calculate_checksum_for_large_object(
+                    s3_uri
+                )
+            ## case: S3 object does not have a SHA256 checksum available
+            elif inventory_row["checksum_algorithm"] != "SHA256":
                 base64_checksum = self.s3_client.generate_checksum_for_object(s3_uri)
+            ## case: S3 object has a SHA256 checksum available
```

Comment:
Hmm, what happens if the file is 5GB _and_ it has a SHA256 checksum available? Can we also **get** the checksum via `S3Client.get_checksum_for_object`? If yes, will need to update the flow of conditions such that it **generates** the checksum only when available.

Reply:
If any object is >= 5gb, then even if the S3 object _claims_ to have an SHA256 checksum, we can't use it.  That's because it's a "COMPOSITE" checksum vs a "FULL" object checksum.  These are easy to spot given a `-###` suffix on them.

This was actually the reasoning behind surfacing the filesize from the inventory data.  If we know the object is >= 5gb then -- no question or other considerations -- we need to calculate the SHA256 in realtime from the actual bytes.

Reply:
But, it's a great question!  Your idea was the first path taken, until realizing this distinction.

Reply:
Ooh, I think this would be great information to include in the docstring for the method!

Reply:
Just saw the docstring you shared above. Might be nice to still include part of this blurb as into the `S3Client` method as well.